### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by stevehobbsdev**
### Description

Here are some minor tweaks to documentation and the readme that I noticed whilst running through the setup myself. See the inline comments for more context about each.

#### Docs updates

Split out the code snippet into two with some intro text, for easier reading:

image

Pulled out the note that was in a code comment into the main body of the text for visibility:

image

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used, if not `master`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,30 +81,26 @@ export default handleAuth();
...
 import { UserProvider } from '@auth0/nextjs-auth0';
 
 export default function App({ Component, pageProps }) {
-  // You can optionally pass the `user` prop from pages that require server side
-  // rendering to prepopulate the `useUser` hook.
-  const { user } = pageProps;
-
   return (
-    <UserProvider user={user}>
+    <UserProvider>
```
Comment (stevehobbsdev) [inline README.md:93]: Pulled this out as it simplifies the initial set up, plus it's repeated in Examples.md anyway. I think it just adds noise here and it's not clear why it's needed.

File: README.md [hunk 2]
```diff
@@ -81,30 +81,26 @@ export default handleAuth();
...
       <Component {...pageProps} />
     </UserProvider>
   );
 }
 ```
 
-Check the user's authentication state and log them in or out from the front end.
+Check whether a user is authenticated by checking that `user` has a value, and log them in or out from the front end by redirecting to the appropriate automatically-generated route.
```
Comment (stevehobbsdev) [inline README.md:100]: Reworded to add clarity on _how_ to check the authenticated state.

File: README.md [hunk 3]
```diff
@@ -81,30 +81,26 @@ export default handleAuth();
 
 This will create the following urls: `/api/auth/login`, `/api/auth/callback`, `/api/auth/logout` and `/api/auth/me`.
 
-Wrap your `pages/_app.jsx` component in the `UserProvider` component.
+Wrap your `pages/_app.js` component in the `UserProvider` component.
 
-```jsx
-// pages/_app.jsx
+```js
+// pages/_app.js
```
Comment (stevehobbsdev) [inline README.md:87]: Renamed these extensions as it seems `.js` is more common, and it's what you get out of the box with a new Next app today.

File: src/config.ts [hunk 4]
```diff
@@ -57,14 +57,20 @@ import { LoginOptions, DeepPartial } from './auth0-session';
...
  * ```js
  * // utils/auth0.js
  * import { initAuth0 } from '@auth0/nextjs-auth0';
  *
  * export default initAuth0({ ...ConfigParameters... });
+ * ```
+ *
+ * Then import it into your route handler:
  *
+ * ```js
```
Comment (stevehobbsdev) [inline src/config.ts:73]: As the code snippet actually deals with code in two files, it's much clearer to have these as two snippets. I just added some intro text to each for context.

File: src/config.ts [hunk 5]
```diff
@@ -108,15 +114,15 @@ export interface Config {
...
    * New values can be passed in to change what is returned from the authorization server
-   * depending on your specific scenario.
+   * depending on your specific scenario. Additional custom parameters can be added as well.
    *
-   * Additional custom parameters can be added as well:
+   * **Note:** You must provide the required parameters if this object is set.
    *
    * ```js
    * {
-   *   // Note: you need to provide required parameters if this object is set.
```
Comment (stevehobbsdev) [inline src/config.ts:117]: This note is much clearer pulled out into the main body of the text (see screenshot in PR for the result)

**Review by adamjmcgrath — APPROVED** [review]

**Pull request merged**